### PR TITLE
Upgrade `tar` to address security vulnerability

### PR DESCRIPTION
**Pull request by iAmmar7**
Upgrade the `tar` dependency to the latest v7.0.1 to address the following security issue:

https://github.com/advisories/GHSA-f5x3-32g6-xq36

### CODE REVIEW

**Comment by cclauss** [comment]
Please add the line `- npm run update-crosswalk` after line 26 of the file `appveyor.yml` so we can see if your tests pass?

File: appveyor.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ install:
   - IF /I "%PLATFORM%" == "x64" CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64
   - IF /I "%PLATFORM%" == "x86" CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x86
   - npm ci
+  - npm run update-crosswalk
```
Comment (cclauss) [inline appveyor.yml:27]: Great!  Now set lines 5 thru 7 of `appveyor.yml` to
```python
    - nodejs_version: 18
    - nodejs_version: 20
    - nodejs_version: 21
```
to match https://nodejs.org/en/about/previous-releases#release-schedule and your tests should be green.  ✅

Comment (iAmmar7) [reply appveyor.yml:27]: The tests would be green because of the newer Node versions, but are you sure I should remove the older versions from the tests?

Comment (iAmmar7) [reply appveyor.yml:27]: Well, it did not pass for Node 20 and 21.

**Comment by bensquire** [comment]
I'm out of my depth here, but just to say we're seeing pressure to also update our packages because of the above tar dependancy.

**Comment by bensquire** [comment]
Thanks @cclauss. Will look at what I need to tear out in turn :)

File: appveyor.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ os: Visual Studio 2015
 
 environment:
   matrix:
-    - nodejs_version: 10
-    - nodejs_version: 12
-    - nodejs_version: 14
+    - nodejs_version: 18
+    - nodejs_version: 20
+    - nodejs_version: 21
```
Comment (benmccann) [inline appveyor.yml:7]: node 22 is out now
```suggestion
    - nodejs_version: 22
```

File: appveyor.yml (hunk 1 above)
Comment (benmccann) [reply appveyor.yml:27]: looks like there's an explanation for that here: https://github.com/mapbox/node-pre-gyp/pull/712/files#diff-92ab9a36df5d8e9f7076f2fdec59492d1ac2d9cf27ea046767a7fc4d542ef3dcR11

**Comment by cclauss** [comment]
Git conflicts -- Please rebase.

**Pull request closed without merging**